CONTRIBUTING.md

```diff
@@ -23,6 +23,20 @@ use GitHub pull requests for this purpose.
 Contributions made by corporations are covered by a different agreement than
 the one above, the Software Grant and Corporate Contributor License Agreement.
 
+### Change log
+After creating a pull request, update the appropriate sections in CHANGES.next.md with a summary of your changes, then commit and push to the branch you are trying to merge.
```

Comment:
seems good, but how are we going to inform contributors about this? during reviews?